File: nomor 2/golang/ex13.go
package main

import (
	"fmt"
	"strconv"
)

func generateData(n int) []string {
	baseNumber := "882"
	var customers []string
	var mobileNumber string
	for i := 0; i < n; i++ {
		mobileNumber = baseNumber
		mobileNumber = fmt.Sprintf("%s%09s", mobileNumber, "0")
		mobileNumber += strconv.Itoa(i)
		customers = append(customers, mobileNumber)
	}
	return customers
}

func sendPromoDiscount(input1 []string, input2 []string) {
	for i := 0; i < len(input1); i++ {
		fmt.Println("Sending Promo to", input1[i])
	}
	fmt.Println("Its Finished to send Promo to", len(input1), "Customers")
	for i := 0; i < len(input2); i++ {
		fmt.Println("Sending Discount to", input2[i])
	}
	fmt.Println("Its Finished to send Discount to", len(input2), "Customers")
}

func main() {
	dataPromo := generateData(100000000)
	dataDiscount := generateData(1000)
	sendPromoDiscount(dataPromo, dataDiscount)
}

File: nomor 2/kotlin/ex13.kt
package main

import java.lang.StringBuilder
import kotlin.random.Random

fun generateData(n: Int): List<String> {
    val baseNumber = "882"
    val customers = mutableListOf<String>()
    val sb = StringBuilder()

    for (i in 0 until n) {
        sb.clear()
        sb.append(baseNumber)
        sb.append("0".repeat(9))
        sb.append(i)
        customers.add(sb.toString())
    }

    return customers
}

fun sendPromoDiscount(input1: List<String>, input2: List<String>) {
    for (i in input1.indices) {
        println("Sending Promo to ${input1[i]}")
    }
    println("Its Finished to send Promo to ${input1.size} Customers")
    for (i in input2.indices) {
        println("Sending Discount to ${input2[i]}")
    }
    println("Its Finished to send Discount to ${input2.size} Customers")
}

fun main() {
    val dataPromo = generateData(100000000)
    val dataDiscount = generateData(1000)
    sendPromoDiscount(dataPromo, dataDiscount)
}
